gmso/utils/singleton.py

```diff
@@ -0,0 +1,7 @@
+class Singleton(object):
+    _inst = None
+
+    def __new__(cls):
+        if cls._inst is None:
+            cls._inst = super(Singleton, cls).__new__(cls)
+        return cls._inst
```

Comment:
Can you explain this to me in plain English? It's a class that returns a super of itself?

Reply:
A singleton class is analogous to a singleton set. As a singleton set can have only one member, a singleton class can have only one object that is created. Regardless of number of calls to the constructor. If you look at the fixture:

https://github.com/umesh-timalsina/gmso/blob/4237bb184d3bc76f66a990e74e046a3d7a428ca3/gmso/tests/test_singleton.py#L5-L8

All the objects created will have same id.

Reply:
You might add a docstring explaining what a singleton class is (a class of which only one instance can exist). See here for some docstring examples that you might adapt. https://stackoverflow.com/a/7346105/2505544